test/access-control.integration.js

```diff
@@ -226,7 +238,101 @@ describe('access control - integration', function() {
+      return '/api/accounts/' + this.accountWithReplaceOnPUTtrue.id;
+    }
+  });
+
+  describe('/accounts with replaceOnPUT false', function() {
+    var count = 0;
+    before(function() {
+      var roleModel = loopback.getModelByType(loopback.Role);
+      roleModel.registerResolver('$dummy', function(role, context, callback) {
+        process.nextTick(function() {
+          if (context.remotingContext) {
+            count++;
+          }
+          callback && callback(null, false); // Always true
+        });
+      });
+    });
+
+    lt.beforeEach.givenModel('accountWithReplaceOnPUTfalse');
+
+    lt.it.shouldBeDeniedWhenCalledAnonymously('GET', '/api/accounts2');
+    lt.it.shouldBeDeniedWhenCalledUnauthenticated('GET', '/api/accounts2');
+    lt.it.shouldBeDeniedWhenCalledByUser(CURRENT_USER, 'GET', '/api/accounts2');
+
+    lt.it.shouldBeDeniedWhenCalledAnonymously('GET', urlForAccount);
+    lt.it.shouldBeDeniedWhenCalledUnauthenticated('GET', urlForAccount);
+    lt.it.shouldBeDeniedWhenCalledByUser(CURRENT_USER, 'GET', urlForAccount);
+
+    lt.it.shouldBeDeniedWhenCalledAnonymously('POST', '/api/accounts2');
+    lt.it.shouldBeDeniedWhenCalledUnauthenticated('POST', '/api/accounts2');
+    lt.it.shouldBeDeniedWhenCalledByUser(CURRENT_USER, 'POST', '/api/accounts2');
+
+    lt.it.shouldBeDeniedWhenCalledAnonymously('POST', urlForReplaceAccountPOST);
+    lt.it.shouldBeDeniedWhenCalledUnauthenticated('POST', urlForReplaceAccountPOST);
+    lt.it.shouldBeDeniedWhenCalledByUser(CURRENT_USER, 'POST', urlForReplaceAccountPOST);
+
+    lt.it.shouldBeDeniedWhenCalledAnonymously('PATCH', urlForAccount);
+    lt.it.shouldBeDeniedWhenCalledUnauthenticated('PATCH', urlForAccount);
+    lt.it.shouldBeDeniedWhenCalledByUser(CURRENT_USER, 'PATCH', urlForAccount);
+
+    lt.describe.whenLoggedInAsUser(CURRENT_USER, function() {
+      beforeEach(function(done) {
+        var self = this;
+        // Create an account under the given user
+        app.models.accountWithReplaceOnPUTfalse.create({
+          userId: self.user.id,
+          balance: 100,
+        }, function(err, act) {
+          self.url = '/api/accounts2/' + act.id;
+
+          done();
+        });
+      });
+
+      lt.describe.whenCalledRemotely('PATCH', '/api/accounts2/:id', function() {
+        lt.it.shouldBeAllowed();
+      });
+
+      lt.describe.whenCalledRemotely('PUT', '/api/accounts2/:id', function() {
+        lt.it.shouldBeAllowed();
+      });
+      lt.describe.whenCalledRemotely('GET', '/api/accounts2/:id', function() {
+        lt.it.shouldBeAllowed();
+      });
+      lt.describe.whenCalledRemotely('DELETE', '/api/accounts2/:id', function() {
+        lt.it.shouldBeDenied();
+      });
+
+      describe('replace on POST verb', function() {
+        beforeEach(function(done) {
+          var self = this;
+          // Create an account under the given user
+          app.models.accountWithReplaceOnPUTfalse.create({
```

Comment:
Could you please find a way how to share the same instance and modify only `this.url`?